src/gpuarray/reduction.h

```diff
@@ -0,0 +1,157 @@
+                                          unsigned        reduxLen,
+                                          const unsigned* reduxList);
+GPUARRAY_PUBLIC int GpuArray_max         (GpuArray*       dst,
+                                          const GpuArray* src,
+                                          unsigned        reduxLen,
+                                          const unsigned* reduxList);
+GPUARRAY_PUBLIC int GpuArray_argmin      (GpuArray*       dstArg,
+                                          const GpuArray* src,
+                                          unsigned        reduxLen,
+                                          const unsigned* reduxList);
+GPUARRAY_PUBLIC int GpuArray_argmax      (GpuArray*       dstArg,
+                                          const GpuArray* src,
+                                          unsigned        reduxLen,
+                                          const unsigned* reduxList);
+GPUARRAY_PUBLIC int GpuArray_minandargmin(GpuArray*       dst,
+                                          GpuArray*       dstArg,
+                                          const GpuArray* src,
+                                          unsigned        reduxLen,
+                                          const unsigned* reduxList);
+GPUARRAY_PUBLIC int GpuArray_maxandargmax(GpuArray*       dst,
+                                          GpuArray*       dstArg,
+                                          const GpuArray* src,
+                                          unsigned        reduxLen,
+                                          const unsigned* reduxList);
+GPUARRAY_PUBLIC int GpuArray_and         (GpuArray*       dst,
+                                          const GpuArray* src,
+                                          unsigned        reduxLen,
+                                          const unsigned* reduxList);
+GPUARRAY_PUBLIC int GpuArray_or          (GpuArray*       dst,
+                                          const GpuArray* src,
+                                          unsigned        reduxLen,
+                                          const unsigned* reduxList);
+GPUARRAY_PUBLIC int GpuArray_xor         (GpuArray*       dst,
+                                          const GpuArray* src,
+                                          unsigned        reduxLen,
+                                          const unsigned* reduxList);
+GPUARRAY_PUBLIC int GpuArray_all         (GpuArray*       dst,
+                                          const GpuArray* src,
+                                          unsigned        reduxLen,
+                                          const unsigned* reduxList);
+GPUARRAY_PUBLIC int GpuArray_any         (GpuArray*       dst,
+                                          const GpuArray* src,
+                                          unsigned        reduxLen,
+                                          const unsigned* reduxList);
```

Comment:
That is true, but it costs in code size and number of exported symbols.

For debugging you can always use conditional breakpoints on the op to only stop for those you are interested in.  I don't really mind making debugging a bit harder here.